src/index.ts

```diff
@@ -597,6 +603,12 @@ prog
       if (report.errorCount) {
         process.exit(1);
       }
+      if (
+        opts['max-warnings'] !== undefined &&
+        report.warningCount > opts['max-warnings']
+      ) {
```

Comment:
```suggestion
      if (report.warningCount > opts['max-warnings']) {
```

if `Infinity` is the default for `max-warnings` then we can simplify this a bit